File: VariableNameRandomGen.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

/* funtion declaration */
char getRandomLowerCaseLetter(); // lowercase letter
char getRandomUpperCaseLetter(); // uppercase letter
char getRandomDigitCharacter(); // digit character
// char getRandomCharacter(); // random character if it will be needed

int main(){
	
	int i = 0;	int r, n; char chr[] = "";
	srand(time(NULL));
	
	printf("Length of the variable? ");
	scanf("%d",&n);
	
	for(i = 0; i<n; i++){
		
		r = rand()%4 + 1;
		
		switch(r){
			case 1:
				chr[0] = getRandomLowerCaseLetter();
				break;
			case 2:
				chr[0] = getRandomUpperCaseLetter();
				break;
			case 3:
				chr[0] = getRandomDigitCharacter();
				break;
			case 4:
				if((i+1)%n == 0) printf("_");
				else printf("_");
				break;
			default: // for any case
				chr[0] = "a";
				break;
		}
		if(r == 4) continue;
		if((i+1)%n == 0) printf("%c \n", chr[0]);
		else printf("%c", chr[0]);	
	}
	return 0;
} // end main

char getRandomLowerCaseLetter(){
	
	char randomletter = 'a' + (rand() % 26);
	return randomletter;
	
}

char getRandomUpperCaseLetter(){
	
	char randomletter = 'A' + (rand() % 26);
	return randomletter;
	
}

char getRandomDigitCharacter(){
	
	char randomletter = '0' + (rand() % 10);
	return randomletter;
	
}
/* this function can be used for random variable generation
char getRandomCharacter(){
	
	char randomletter = rand() % 129; // after 129 they can not be display
	return randomletter;
	
} */

File: VariableNameRandomGen.java
import java.util.Scanner;
public class VariableNameRandomGen{
	
	public static void main(String args[]){
		int n, r;  char chr = 'a';
		Scanner sc = new Scanner(System.in);
		
		System.out.print("Length of the variable? ");
		n = sc.nextInt();
		
		for(int i = 0; i<n;i++){ 
						
			r = (int)(Math.random()*4+1); /* generate random value between 1-5 */
			switch(r){
				case 1:
					chr = getRandomLowerCaseLetter();
					break;
				
				case 2:
					chr = getRandomUpperCaseLetter();
					break;
				
				case 3:
					chr = getDigitCharacter();
					break;
				
				case 4:
					if ((int)(Math.random()*2+1) == 1) chr = '_';
					else chr = '$';
					break;
				
				default: // do not do any operation
					break;
				
			}
			
			if((i+1)%n==0) System.out.println(chr);
			else System.out.print(chr);
		}
	}
	
	public static char getRandomCharacter(char ch1, char ch2){
		return (char)(ch1 + Math.random()*(ch2 - ch1 + 1));
	}
	
	/** Genetare a random lowercase letter */
	public static char getRandomLowerCaseLetter(){
		return getRandomCharacter('a','z');
	}
	
	/** Generate a random uppercase letter */
	public static char getRandomUpperCaseLetter(){
		return getRandomCharacter('A','Z');
	}
	
	/** Generate a random digit character */
	public static char getDigitCharacter(){
		return getRandomCharacter('0','9');
	}
	
	/** This fuction can be used for generation character between '_' and '$' 
	public static char getRandomOtherCharacters(){
		return getRandomCharacter('_','$');
	} */
} // end of class